daemon/info.go

```diff
@@ -16,7 +16,7 @@ import (
 	"github.com/docker/docker/utils"
 )
 
-func (daemon *Daemon) SystemInfo() (*types.Info, error) {
+func (daemon *Daemon) SystemInfo(includeOperatingSystem bool) (*types.Info, error) {
```

Comment:
@olleolleolle variable name is confusing. It should have the word `legacy` or `old`... `useLegacyOperatingSystemField`, idk whatever that would not suggest that if `includeOperatingSystem` is `false`, then `OS` (the new field) won't be included.